include/OpenSpaceToolkit/Astrodynamics/Flight/System/PropulsionSystem.hpp

```diff
@@ -0,0 +1,187 @@
+    ///
+    /// @code
+    ///                     PropulsionSystem propulsion = { ... };
+    /// @endcode
+    ///
+    /// @param              [in] aThruster Thruster (real)
+    /// @param              [in] aSpecificImpulse Specific impulse
+
+    PropulsionSystem(const Real& aThruster, const Real& aSpecificImpulse);
+
+    /// @brief              Equal to operator
+    ///
+    /// @param              [in] aPropulsionSystem A propulsion system
+    /// @return             True if propulsion systems are equal
+
+    bool operator==(const PropulsionSystem& aPropulsionSystem) const;
+
+    /// @brief              Not equal to operator
+    ///
+    /// @param              [in] aPropulsionSystem A propulsion system
+    /// @return             True if propulsion systems are not equal
+
+    bool operator!=(const PropulsionSystem& aPropulsionSystem) const;
+
+    /// @brief              Output stream operator
+    ///
+    /// @param              [in] anOutputStream An output stream
+    /// @param              [in] aPropulsionSystem A propulsion system
+    /// @return             A reference to output stream
+
+    friend std::ostream& operator<<(std::ostream& anOutputStream, const PropulsionSystem& aPropulsionSystem);
+
+    /// @brief              Check if propulsion system is defined
+    ///
+    /// @return             True if propulsion system is defined
+
+    bool isDefined() const;
+
+    /// @brief              Print propulsion system
+    ///
+    /// @param              [in] anOutputStream An output stream
+    /// @param              [in] (optional) displayDecorators If true, display decorators
+
+    void print(std::ostream& anOutputStream, bool displayDecorator = true) const;
+
+    /// @brief              Get propulsion system's thrust
+    ///
+    /// @code
+    ///                     Scalar thrust = propulsionSystem.getThrust() ;
+    /// @endcode
+    ///
+    /// @return             Scalar
+
+    Scalar getThrust() const;
+
+    /// @brief              Get propulsion system's specific impulse
+    ///                     https://en.wikipedia.org/wiki/Specific_impulse
+    ///
+    /// @code
+    ///                     Scalar specificImpulse = propulsionSystem.getSpecificImpulse() ;
+    /// @endcode
+    ///
+    /// @return             Scalar
+
+    Scalar getSpecificImpulse() const;
+
+    /// @brief              Get propulsion system's mass flow rate
+    ///
+    /// @code
+    ///                     Scalar massFlowRate = propulsionSystem.getMassFlowRate() ;
+    /// @endcode
+    ///
+    /// @return             Scalar
+
+    Scalar getMassFlowRate() const;
+
+    /// @brief              Get propulsion system's acceleration
+    ///
+    /// @code
+    ///                     Real acceleration = propulsionSystem.getAcceleration() ;
+    /// @endcode
+    ///
+    /// @return             Real
+
+    Real getAcceleration(const Mass& aMass) const; // TBI: Should return acceleration unit
```

Comment:
It might be worth having both, as this will be used in the dynamics class where we don't want to convert between units often, for performance.